snuba/migrations/snuba_migrations/functions/0001_functions.py

```diff
@@ -0,0 +1,189 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    AggregateFunction,
+    Array,
+    Column,
+    DateTime,
+    Float,
+    Nested,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+common_columns: List[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("transaction_name", String()),
+    Column("timestamp", DateTime()),
+    Column("depth", UInt(32)),
+    Column("parent_fingerprint", UInt(64)),
+    Column("fingerprint", UInt(64)),
+    Column("symbol", String()),
+    Column("image", String()),
+    Column("filename", String()),
+    Column("is_application", UInt(8)),
+    Column("tags", Nested([Column("key", String()), Column("value", String())])),
```

Comment:
Why do we need custom arbitrary tags? These are not a good fit for rollup scenarios as a customer could be able to multiply the size of your table by just adding new tags.

Reply:
For ios/android, this will largely be a few device fields like
- device os name/version
- release
- android API level
These are mostly system generated, but release is user generated. In practice, this should be fairly low cardinality but we will verify this as we see more data.

Reply:
In the end it will be your choice, though I would discourage pretty strongly tags in aggregating merge trees. 
That tends to generate wild variance in your storage size and query performance.
Let's say that you add a simple tag to all your messages somewhere early in the product without thinking about that too much -> you just doubled your storage size in one shot as, from that point on, you store multiple buckets when you were storing one.

If you know which tags you want, just add columns for them

Reply:
Understood, this is definitely something that we'll keep in mind. I think what we'll do here is start with these tag columns for the flexibility right now, and tread carefully. Once we have a clearer understanding of the right fields to keep for the various platforms, we'll work on adding columns for them.

Reply:
This has been removed in favour of a few columns that we know we want.